string name
uint8 location
---
string result